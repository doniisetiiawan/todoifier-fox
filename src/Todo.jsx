import PropTypes from 'prop-types';
import React, { Component } from 'react';
import './Todo.css';

class Todo extends Component {
  static propTypes = {
    description: PropTypes.string.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      description: props.description,
      done: false,
    };
  }

  markAsDone = () => {
    this.setState({ done: true });
  };

  cssClasses = () => {
    let classes = ['Todo'];
    const { done } = this.state;
    if (done) {
      classes = [...classes, 'Done'];
    }
    return classes.join(' ');
  };

  render() {
    const { description } = this.state;
    return (
      <div className={this.cssClasses()}>
        {description}
        <br />
        <button
          type="button"
          onClick={this.markAsDone}
        >Mark as Done
        </button>
      </div>
    );
  }
}

export default Todo;
