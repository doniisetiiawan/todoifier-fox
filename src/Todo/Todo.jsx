/* eslint-disable react/require-default-props */
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import styles from './Todo.module.css';

class Todo extends Component {
  static propTypes = {
    description: PropTypes.string,
  };

  constructor(props) {
    super(props);
    this.state = {
      description: props.description,
      done: false,
      critical: false,
    };
  }

  markAsDone = () => {
    this.setState({ done: true });
  };

  cssClasses = () => {
    let classes = [];
    const { critical, done } = this.state;
    if (critical) {
      classes = [styles.critical];
    } else {
      classes = [styles.todo];
    }
    if (done) {
      classes = [...classes, styles.done];
    }
    return classes.join(' ');
  };

  removeTodo = () => {
    // eslint-disable-next-line react/prop-types
    const { removeTodo: removeTodo1, description } = this.props;
    removeTodo1(description);
  };

  markCritical = () => {
    this.setState({ critical: true });
  };

  render() {
    const { description } = this.state;
    return (
      <div className={this.cssClasses()}>
        {description}
        <hr className={styles.redDivider} />
        <button
          type="button"
          className="MarkDone"
          onClick={this.markAsDone}
        >Mark as Done
        </button>
        <button
          type="button"
          className="RemoveTodo"
          onClick={this.removeTodo}
        >Remove Me
        </button>
        <button
          type="button"
          className="MarkCritical"
          onClick={this.markCritical}
        >Mark as Critical
        </button>
      </div>
    );
  }
}

export default Todo;
