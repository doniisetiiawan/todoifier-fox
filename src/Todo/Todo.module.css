.todo {
    border: 2px solid black;
    text-align: center;
    background: #f5f5f5;
    color: #333;
    margin: 20px;
    padding: 20px;
}

.done {
    background: #f5a5a5;
}

.redDivider {
    border: 2px solid red;
}

.critical {
    composes: todo;
    border: 4px dashed red;
}
