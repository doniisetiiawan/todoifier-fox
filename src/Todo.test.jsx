import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Todo from './Todo';

describe(Todo, () => {
  const description = 'New Todo';
  const mockRemoveTodo = jest.fn();
  const component = shallow(
    <Todo description={description} removeTodo={mockRemoveTodo} />,
  );
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Todo />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders and matches our snapshot', () => {
    // eslint-disable-next-line no-shadow
    const component = renderer.create(<Todo description="Yo" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
