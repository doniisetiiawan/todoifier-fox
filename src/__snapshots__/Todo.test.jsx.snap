// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders and matches our snapshot 1`] = `
<div
  className="Todo"
>
  Yo
  <br />
  <button
    onClick={[Function]}
    type="button"
  >
    Mark as Done
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Remove Me
  </button>
</div>
`;
